packages/pangea-sdk/pangea/asyncio/request.py

```diff
@@ -158,6 +173,90 @@ async def put_presigned_url(self, url: str, files: List[Tuple]):
+                    "action": "download_file",
+                    "url": url,
+                    "filename": filename,
+                    "folder": dest_folder,
+                    "status": "start",
+                }
+            )
+        )
+        async with self.session.get(url, headers={}) as response:
+            if response.status == 200:
+                if filename is None:
+                    content_disposition = response.headers.get("Content-Disposition", "")
+                    filename = self._get_filename_from_content_disposition(content_disposition)
+                    if filename is None:
+                        filename = self._get_filename_from_url(url)
+                        if filename is None:
+                            filename = "download_file"
+
+                if dest_folder is None:
+                    dest_folder = "./"
+
+                destination_path = os.path.join(dest_folder, filename)
+
+                directory = os.path.dirname(destination_path)
+                if not os.path.exists(directory):
+                    os.makedirs(directory)
+
+                with open(destination_path, "wb") as file:
+                    file.write(await response.read())
+
+                self.logger.debug(
+                    json.dumps(
+                        {
+                            "service": self.service,
+                            "action": "download_file",
+                            "url": url,
+                            "filename": filename,
+                            "folder": dest_folder,
+                            "status": "success",
+                        }
+                    )
+                )
+            else:
+                raise pe.DownloadFileError(f"Failed to download file. Status: {response.status}", await response.text())
+
+    async def _get_pangea_json(self, reader: aiohttp.MultipartReader) -> Optional[Dict]:
+        # Iterate through parts
+        async for part in reader:
+            return await part.json()
+        return None
+
+    async def _get_attached_files(self, reader: aiohttp.MultipartReader) -> List[AttachedFile]:
+        files = []
+        i = 0
+
+        async for part in reader:
+            content_type = part.headers.get("Content-Type", "")
+            content_disposition = part.headers.get("Content-Disposition", "")
+            name = self._get_filename_from_content_disposition(content_disposition)
+            if name is None:
+                name = f"default_file_name_{i}"
+                i += 1
+            files.append(AttachedFile(name, await part.read(), content_type))
+
+        return files
+
+    async def _process_multipart_response(self, resp: aiohttp.ClientResponse) -> MultipartResponse:
+        # Parse the multipart response
+        multipart_reader = aiohttp.MultipartReader.from_response(resp)
+
+        pangea_json = await self._get_pangea_json(multipart_reader)  # type: ignore
```

Comment:
At least use a more specific ignore
```suggestion
        pangea_json = await self._get_pangea_json(multipart_reader)  # type: ignore[arg-type]
```
This applies to the below ones as well.

Reply:
Fixed